parser/src/parser/tests/mod.rs

```diff
@@ -1,18 +1,619 @@
+            access_type: AccessType::Index($idx),
+            offset: $offset,
+            ty: Some($ty),
+        })
+    };
+
+    ($name:literal [ $idx:literal ], $offset:literal) => {
+        ScalarExpr::SymbolAccess(SymbolAccess::new(
+            miden_diagnostics::SourceSpan::UNKNOWN,
+            ident!($name),
+            AccessType::Index($idx),
+            $offset,
+        ))
+    };
+
+    ($module:ident, $name:ident [ $idx:literal ], $offset:literal, $ty:expr) => {
+        ScalarExpr::SymbolAccess(SymbolAccess {
+            span: miden_diagnostics::SourceSpan::UNKNOWN,
+            name: ResolvableIdentifier::Resolved(ident!($module, $name)),
+            access_type: AccessType::Index($idx),
+            offset: $offset,
+            ty: Some($ty),
+        })
+    };
+}
+
+macro_rules! expr {
+    ($expr:expr) => {
+        Expr::try_from($expr).unwrap()
+    };
+}
+
+macro_rules! slice {
+    ($name:ident, $range:expr) => {
+        ScalarExpr::SymbolAccess(SymbolAccess {
+            span: miden_diagnostics::SourceSpan::UNKNOWN,
+            name: ResolvableIdentifier::Unresolved(NamespacedIdentifier::Binding(ident!($name))),
+            access_type: AccessType::Slice($range),
+            offset: 0,
+            ty: None,
+        })
+    };
+
+    ($name:ident, $range:expr, $ty:expr) => {
+        ScalarExpr::SymbolAccess(SymbolAccess {
+            span: miden_diagnostics::SourceSpan::UNKNOWN,
+            name: ResolvableIdentifier::Local(ident!($name)),
+            access_type: AccessType::Slice($range),
+            offset: 0,
+            ty: Some($ty),
+        })
+    };
+}
+
+macro_rules! bounded_access {
+    ($name:ident, $bound:expr) => {
+        ScalarExpr::BoundedSymbolAccess(BoundedSymbolAccess::new(
+            miden_diagnostics::SourceSpan::UNKNOWN,
+            SymbolAccess::new(miden_diagnostics::SourceSpan::UNKNOWN, ident!($name), AccessType::Default, 0),
+            $bound,
+        ))
+    };
+
+    ($name:ident, $bound:expr, $ty:expr) => {
+        ScalarExpr::BoundedSymbolAccess(BoundedSymbolAccess::new(
+            miden_diagnostics::SourceSpan::UNKNOWN,
+            SymbolAccess {
+                span: miden_diagnostics::SourceSpan::UNKNOWN,
+                name: ResolvableIdentifier::Local(ident!($name)),
+                access_type: AccessType::Default,
+                offset: 0,
+                ty: Some($ty),
+            },
+            $bound,
+        ))
+    };
+
+    ($module:ident, $name:ident, $bound:expr, $ty:expr) => {
+        ScalarExpr::BoundedSymbolAccess(BoundedSymbolAccess::new(
+            miden_diagnostics::SourceSpan::UNKNOWN,
+            SymbolAccess {
+                span: miden_diagnostics::SourceSpan::UNKNOWN,
+                name: ResolvableIdentifier::Resolved(ident!($module, $name)),
+                access_type: AccessType::Default,
+                offset: 0,
+                ty: Some($ty),
+            },
+            $bound,
+        ))
+    };
```

Comment:
similar question/comment to previously about only trace columns only being defined in the root module. There's another case below as well

Reply:
Yep, this can go away. I created a lot of these upfront before I knew which ones I'd actually need - I was anticipating that I might need to create invalid structures in some cases, but that didn't really happen, so a number of these macros have cases that can be eliminated.